tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/source_files/source_files_container_test.ts

```diff
@@ -0,0 +1,237 @@
+import {TestBed} from '@angular/core/testing';
+import {By} from '@angular/platform-browser';
+
+import {Store} from '@ngrx/store';
+import {provideMockStore, MockStore} from '@ngrx/store/testing';
+
+import {DebuggerComponent} from '../../debugger_component';
+import {DebuggerContainer} from '../../debugger_container';
+import {DataLoadState, State} from '../../store/debugger_types';
+import {createDebuggerState, createState} from '../../testing';
+import {AlertsModule} from '../alerts/alerts_module';
+import {ExecutionDataModule} from '../execution_data/execution_data_module';
+import {InactiveModule} from '../inactive/inactive_module';
+import * as loadMonacoShim from '../source_code/load_monaco_shim';
+import {StackTraceModule} from '../stack_trace/stack_trace_module';
+import {
+  getFocusedSourceFileContent,
+  getFocusedSourceLineSpec,
+} from '../../store';
+import {TimelineModule} from '../timeline/timeline_module';
+import {SourceFilesContainer} from './source_files_container';
+import {SourceFilesModule} from './source_files_module';
+
+/** @typehack */ import * as _typeHackStore from '@ngrx/store';
+
+describe('Source Files Container', () => {
+  let store: MockStore<State>;
+  let dispatchSpy: jasmine.Spy;
+
+  class FakeRange {
+    constructor(
+      readonly startLineNumber: number,
+      readonly startColumn: number,
+      readonly endLineNumber: number,
+      readonly endColumn: number
+    ) {}
+  }
+
+  let loadMonacoSpy: jasmine.Spy;
+  // TODO(cais): Explore better typing by depending on 3rd-party libraries.
+  let editorSpy: jasmine.SpyObj<any>;
+  let monaco: any;
+  function setUpMonacoFakes() {
+    async function fakeLoadMonaco() {
+      monaco = {
+        editor: {
+          create: () => {
+            editorSpy = jasmine.createSpyObj('editorSpy', [
+              'deltaDecorations',
+              'revealLineInCenter',
+              'setValue',
+            ]);
+            return editorSpy;
+          },
+          ScrollType: {
+            Immediate: 1,
+            Smooth: 0,
+          },
+        },
+        Range: FakeRange,
+      };
+      loadMonacoShim.windowWithRequireAndMonaco.monaco = monaco;
+    }
+    loadMonacoSpy = spyOn(loadMonacoShim, 'loadMonaco').and.callFake(
+      fakeLoadMonaco
+    );
+  }
+
+  function tearDownMonacoFakes() {
+    if (loadMonacoShim.windowWithRequireAndMonaco.monaco !== undefined) {
+      delete loadMonacoShim.windowWithRequireAndMonaco.monaco;
+    }
+  }
+
+  beforeEach(async () => {
+    setUpMonacoFakes();
+    await TestBed.configureTestingModule({
+      declarations: [DebuggerComponent, DebuggerContainer],
+      imports: [
+        AlertsModule,
+        CommonModule,
+        ExecutionDataModule,
+        InactiveModule,
+        SourceFilesModule,
+        StackTraceModule,
+        TimelineModule,
+      ],
+      providers: [
+        provideMockStore({
+          initialState: createState(createDebuggerState()),
+        }),
```

Comment:
FYI, if you are going to use `overrideSelector` pattern, you don't need to pass the initialState (though I would make sure to set the selectors' values to something reasonable in beforeEach)